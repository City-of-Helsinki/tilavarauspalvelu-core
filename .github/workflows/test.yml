name: "Tests"

on:
  push:
    branches:
      - main
  pull_request:
  workflow_dispatch:

# Cancel a workflow from the same PR, branch or tag when a new workflow is triggered.
concurrency:
  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
  cancel-in-progress: true

jobs:

  test:
    name: "Run tests and SonarCloud scan"
    runs-on: ubuntu-latest

    services:
      postgres:
        image: postgis/postgis:13-3.3-alpine
        env:
          POSTGRES_USER: tvp
          POSTGRES_PASSWORD: tvp
          POSTGRES_DB: tvp
          DEBUG: true
          SECRET_KEY: build_secret
        # Set health checks to wait until postgres has started
        options: >-
          --health-cmd pg_isready
          --health-interval 10s
          --health-timeout 5s
          --health-retries 5
        ports:
          - 5432:5432

      redis:
        image: redis:7-alpine
        # Set health checks to wait until redis has started
        options: >-
          --health-cmd "redis-cli ping"
          --health-interval 10s
          --health-timeout 5s
          --health-retries 5
        ports:
          - 6379:6379

      elastic:
        image: docker.elastic.co/elasticsearch/elasticsearch:8.8.0
        env:
          xpack.security.enabled: false
          discovery.type: single-node
        ports:
          - 9200:9200

    steps:
      - name: "Checkout code"
        uses: actions/checkout@v3
        with:
          # Disable shallow cloning to improving relevancy of reporting for SonarCloud
          fetch-depth: 0

      - name: "Install poetry"
        run: pipx install "poetry==1.6.0"

      - name: "Install GDAL"
        run: sudo apt install -y --no-install-recommends gdal-bin

      - name: "Set up python"
        uses: actions/setup-python@v4
        with:
          cache: "poetry"
          python-version: "3.11"

      - name: "Install dependencies"
        run: poetry install

      - name: "Run pytest with coverage"
        run: poetry run pytest --cov=. --cov-report=xml --cov-branch
        env:
          CACHE_LOCATION: redis://localhost:6379/0
          DATABASE_URL: postgis://tvp:tvp@localhost/tvp
          DEBUG: true
          SECRET_KEY: build_secret

      # Without this workaround Sonar reports a warning about an incorrect source path; credits to linked-events
      - name: "Override coverage report source path for Sonar"
        run: sed -i 's@'$GITHUB_WORKSPACE'@/github/workspace/@g' coverage.xml

      - name: "SonarCloud Scan"
        uses: SonarSource/sonarcloud-github-action@master
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
