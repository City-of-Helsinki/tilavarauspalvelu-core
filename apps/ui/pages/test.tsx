import CustomErrorComponent from "@/pages/_error";
import type { GetServerSidePropsContext } from "next";
import { getCommonServerSideProps } from "@/modules/serverUtils";
import { serverSideTranslations } from "next-i18next/serverSideTranslations";
import { useSearchParams } from "next/navigation";

function Page() {
  const params = useSearchParams();
  const statusCode = params.get("statusCode") ? Number(params.get("statusCode")) : 500;
  return <CustomErrorComponent statusCode={statusCode} />;
}

export async function getStaticProps({ locale }: GetServerSidePropsContext) {
  return {
    props: {
      ...getCommonServerSideProps(),
      ...(await serverSideTranslations(locale ?? "fi")),
    },
  };
}

export default Page;
