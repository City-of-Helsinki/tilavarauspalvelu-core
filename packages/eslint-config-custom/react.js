// @ts-check
const { defineConfig } = require('eslint-define-config');
const { resolve } = require('node:path');

const project = resolve(process.cwd(), "tsconfig.json");

const SCHEMA_PATH = "../../tilavaraus.graphql";

/// <reference types="@eslint-types/typescript-eslint" />
module.exports = defineConfig({
  overrides: [
    {
      "files": ["*.ts", "*.tsx"],
      "processor": "@graphql-eslint/graphql",
    },
    {
      "files": ["*.graphql"],
      "parser": "@graphql-eslint/eslint-plugin",
      "plugins": ["@graphql-eslint"],
      // operations is the client side queries
      "extends": "plugin:@graphql-eslint/operations-recommended",
      parserOptions: {
        schema: SCHEMA_PATH,
        // monorepo paths... should use a find root function
        operations: ["../../apps/**/*.{ts,tsx}", "../../packages/**/*.{ts,tsx}"],
        // our graphql codegen config is old and has issues disable it and use the above
        skipGraphQLConfig: true
     },
      rules: {
        "@graphql-eslint/no-deprecated": "warn",
      },
    },
  ],
  root: true,
  extends: [
    require.resolve('@vercel/style-guide/eslint/node'),
    require.resolve('@vercel/style-guide/eslint/typescript'),
    require.resolve('@vercel/style-guide/eslint/browser'),
    require.resolve('@vercel/style-guide/eslint/react'),
    require.resolve('@vercel/style-guide/eslint/jest-react'),
    "plugin:prettier/recommended",
  ],
  plugins: [
    "prettier",
  ],
  parserOptions: {
    project,
  },
  settings: {
    'import/resolver': {
      typescript: {
        project,
      },
    },
  },
  ignorePatterns: [
    'gql-types.ts',
  ],
  env: {
    browser: true,
    jest: true,
    node: true,
  },
  globals: {
    React: true,
    JSX: true,
  },
  rules: {
    // TODO temp rules from migrating from airbnb to vercel style-guide
    "@typescript-eslint/prefer-nullish-coalescing": 0,
    "unicorn/filename-case": 0,
    "@typescript-eslint/no-unsafe-enum-comparison": 0,
    "@typescript-eslint/explicit-function-return-type": 0,
    "react/jsx-sort-props": 0,
    "react/jsx-no-leaked-render": 0,
    "@typescript-eslint/no-unnecessary-condition": 0,
    "@typescript-eslint/no-confusing-void-expression": 0,
    "eslint-comments/require-description": 0,
    "@typescript-eslint/no-misused-promises": 0,
    "@typescript-eslint/no-unsafe-member-access": 0,
    "@typescript-eslint/consistent-type-definitions": 0,
    "@typescript-eslint/consistent-type-imports": 0,
    "@typescript-eslint/require-await": 0,
    "@typescript-eslint/no-floating-promises": 0,
    "@typescript-eslint/prefer-optional-chain": 0,
    "@typescript-eslint/no-unsafe-assignment": 0,
    "@typescript-eslint/prefer-regexp-exec": 0,
    "@typescript-eslint/non-nullable-type-assertion-style": 0,
    "@typescript-eslint/array-type": 0,
    "@typescript-eslint/prefer-ts-expect-error": 0,
    "@typescript-eslint/no-unsafe-argument": 0,
    "@typescript-eslint/restrict-template-expressions": 0,
    "@typescript-eslint/consistent-indexed-object-style": 0,
    "@typescript-eslint/consistent-type-assertions": 0,
    "@typescript-eslint/no-unnecessary-type-assertion": 0,
    "@typescript-eslint/no-unsafe-call": 0,
    "@typescript-eslint/no-unsafe-return": 0,
    "@typescript-eslint/no-unnecessary-type-arguments": 0,
    // TODO there should be rules for this already
    "@typescript-eslint/naming-convention": 0,
    "no-implicit-coercion": 0,
    "import/no-named-as-default-member": 0,
    "import/no-named-as-default": 0,
    "import/no-default-export": 0,
    "import/order": 0,
    "tsdoc/syntax": 0,
    "jest/require-top-level-describe": 0,
    "jest/no-conditional-expect": 0,
    "jest/prefer-lowercase-title": 0,
    "jest/no-disabled-tests": 0,
    "jest/no-identical-title": 0,
    "jest/no-mocks-import": 0,
    "testing-library/prefer-screen-queries": 0,
    "testing-library/render-result-naming-convention": 0,
    "testing-library/prefer-find-by": 0,
    "testing-library/prefer-presence-queries": 0,
    "testing-library/await-async-events": 0,
    "testing-library/await-async-utils": 0,
    "testing-library/no-node-access": 0,
    "testing-library/no-wait-for-multiple-assertions": 0,
    "react/display-name": 0,
    "react/hook-use-state": 0,
    "react/jsx-pascal-case": 0,
    "camelcase": 0,
    "no-constant-binary-expression": 0,
    "prefer-named-capture-group": 0,
    // fails to find lodash...
    "import/named": 0,
    // afaik this conflicts with @typescript-eslint/no-unused-vars
    "no-unused-vars": 0,
    // enforce null checks are == / != everything else ===
    eqeqeq: ["error", "always", { null: "never" }],
    // mixing mjs with ts in nextjs causes issues
    "import/extensions": [
      "error",
      "never",
      {
        "mjs": "always",
        "cjs": "always",
        "json": "always",
      },
    ],
    // prefer explicit cases over defaults
    "default-case": 0,
    // FIXME this is missing
    // "@typescript-eslint/switch-exhaustiveness-check": "error",
    "import/no-extraneous-dependencies": [
      "error",
      {
        devDependencies: true,
      },
    ],
    // allow deconstruction and _ for unused vars
    "@typescript-eslint/no-unused-vars": [
      "error",
      {
        varsIgnorePattern: "^_",
        argsIgnorePattern: "^_",
        ignoreRestSiblings: true,
      },
    ],
    "import/no-unresolved": "off",
    "import/prefer-default-export": 0,
    "jsx-a11y/alt-text": 1,
    "jsx-a11y/label-has-associated-control": [
      "error",
      {
        labelComponents: [],
        labelAttributes: [],
        controlComponents: [],
        assert: "htmlFor",
        depth: 25,
      },
    ],
    "no-nested-ternary": 0,
    "react/jsx-no-useless-fragment": [
      "error",
      {
        allowExpressions: true,
      },
    ],
    "react/destructuring-assignment": 0,
    "react/jsx-props-no-spreading": 0,
    "react/prop-types": 0,
    "react/require-default-props": 0,
    "react/static-property-placement": 0,
    "react/function-component-definition": [
      "error",
      {
        namedComponents: ["function-declaration", "arrow-function"],
        unnamedComponents: ["function-expression", "arrow-function"],
      },
    ],
    // There seems to be no other way to override this than disabling it and rewriting the rules in the naming-convention
    "@typescript-eslint/camelcase": ["off"],
    "@typescript-eslint/no-empty-function": 0,
    "react/no-unstable-nested-components": [
      "warn",
      {
        allowAsProps: true,
      },
    ],
  },
});
