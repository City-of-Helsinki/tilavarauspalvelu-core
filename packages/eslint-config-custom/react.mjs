// @ts-check
import eslint from "@eslint/js";
import tseslint from "typescript-eslint";
import globals from "globals";
import reactPlugin from "eslint-plugin-react";
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore -- types missing?
import eslintPluginReactHooks from "eslint-plugin-react-hooks";
import eslintPluginPrettierRecommended from "eslint-plugin-prettier/recommended";
import graphqlPlugin from "@graphql-eslint/eslint-plugin";
import eslintPluginImportX from "eslint-plugin-import-x";

const SCHEMA_FILE = "../../tilavaraus.graphql";

/** Construct a graphql config and only include that project's gql files
 *  @param path {string}
 *  @param disableUnusedFragments {boolean}
 * */
export function constructGQLConfig(path, disableUnusedFragments = false) {
  return {
    files: ["**/*.graphql"],
    languageOptions: {
      parser: graphqlPlugin.parser,
      parserOptions: {
        graphQLConfig: {
          schema: SCHEMA_FILE,
          documents: [`${path}/**/!(*.d|gql-types).{ts,tsx}`],
        },
      },
    },
    plugins: {
      "@graphql-eslint": graphqlPlugin,
    },
    rules: {
      ...graphqlPlugin.configs["flat/operations-recommended"].rules,
      "@graphql-eslint/selection-set-depth": ["error", { maxDepth: 10 }],
      // can't use shared fragments in common if this rule is on
      // disabling rules inside gql doesn't work sometimes
      "@graphql-eslint/no-unused-fragments": disableUnusedFragments
        ? "off"
        : "error",
    },
  };
}

/** @type {import('eslint').Linter.RulesRecord} */
const myRules = {
  "no-console": "error",
  // enforce null checks are == / != everything else ===
  eqeqeq: ["error", "always", { null: "never" }],
  // afaik this conflicts with @typescript-eslint/no-unused-vars
  "no-unused-vars": 0,
  // allow deconstruction and _ for unused vars
  "@typescript-eslint/no-unused-vars": [
    "error",
    {
      args: "all",
      argsIgnorePattern: "^_",
      varsIgnorePattern: "^_|^React$",
      caughtErrors: "all",
      caughtErrorsIgnorePattern: "^_",
      destructuredArrayIgnorePattern: "^_",
      ignoreRestSiblings: true,
    },
  ],
  "@typescript-eslint/no-non-null-assertion": "error",
  // this only catches await errors if there is a single await that is missing
  // more complex rules:
  // - no-misused-promises
  // - no-floating-promises
  // both in typescript-eslint/recommended-type-checked
  // but out type generation is broken (so only recommended works)
  "require-await": "error",
  // for some reason this raises errors for "key" props in jsx arrays
  "react/prop-types": "off",
  "react/no-array-index-key": "error",
  "react/no-danger": "error",
  "react/no-invalid-html-attribute": "error",
  "react/jsx-wrap-multilines": "error",
  "react/jsx-no-useless-fragment": "error",
  "react/jsx-no-script-url": "error",
  "react/jsx-boolean-value": "error",
  "react/forward-ref-uses-ref": "error",
  "react/button-has-type": "error",
  "react/no-this-in-sfc": "error",
  "react/prefer-stateless-function": "error",
  "react/void-dom-elements-no-children": "error",
  // TODO enable in a separate commit
  // "react/no-unstable-nested-components": "error",
  // styled-components doesn't work with this (alternative would be disable this only for styled-components)
  "import-x/no-named-as-default": "off",
  // breaks admin-ui typescript paths
  "import-x/no-unresolved": "off",
  "import-x/no-dynamic-require": "warn",
  // mixing mjs with ts in nextjs causes issues
  "import-x/extensions": [
    "error",
    "never",
    {
      mjs: "always",
      cjs: "always",
      json: "always",
    },
  ],
};

const myConfig = {
  ...reactPlugin.configs.flat.recommended,
  processor: graphqlPlugin.processor,
  settings: {
    react: {
      version: "detect",
    },
  },
  languageOptions: {
    ...reactPlugin.configs.flat.recommended?.languageOptions,
    ecmaVersion: 2022,
    sourceType: "module",
    parserOptions: {
      ecmaFeatures: {
        jsx: true,
      },
    },
    globals: {
      ...globals.browser,
      ...globals.node,
    },
  },
  plugins: {
    react: reactPlugin,
  },
  rules: myRules,
};

const tsConfig = tseslint.config(
  eslint.configs.recommended,
  tseslint.configs.recommended,
);

const ignores = [
  "node_modules/",
  ".next/",
  ".turbo/",
  "out/",
  "public/",
  "**/gql-types.ts",
];

export default [
  ...tsConfig,
  reactPlugin.configs.flat["jsx-runtime"],
  {
    plugins: {
      "react-hooks": eslintPluginReactHooks,
    },
    rules: { ...eslintPluginReactHooks.configs.recommended.rules },
  },
  eslintPluginPrettierRecommended,
  eslintPluginImportX.flatConfigs.recommended,
  eslintPluginImportX.flatConfigs.typescript,
  myConfig,
  {
    ignores,
  },
];
