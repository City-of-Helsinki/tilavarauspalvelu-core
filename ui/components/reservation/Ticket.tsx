import React from "react";
import { differenceInMinutes, parseISO } from "date-fns";
import { useTranslation } from "react-i18next";
import {
  getPriceUnitMinutes,
  getReservationPrice,
  getReservationVolume,
} from "common";
import styled from "styled-components";
import { IconCalendar } from "hds-react";
import { isFinite } from "lodash";
import { i18n } from "next-i18next";
import { fontMedium, H1, H2, Strongish } from "common/src/common/typography";
import { breakpoints } from "common/src/common/style";
import { capitalize, formatDurationMinutes } from "../../modules/util";
import {
  ReservationUnitByPkType,
  ReservationUnitsReservationUnitPriceUnitChoices,
} from "../../modules/gql-types";
import { getPrice } from "../../modules/reservationUnit";
import { NoWrap } from "../../styles/util";

export type TicketState = "incomplete" | "complete" | "error";

type Props = {
  state: TicketState;
  title: string;
  subtitle?: string;
  begin?: string;
  end?: string;
  isFree?: boolean;
  bgColor?: string;
  reservationPrice?: number;
  lowestPrice?: number;
  highestPrice?: number;
  priceUnit?: ReservationUnitsReservationUnitPriceUnitChoices;
  taxPercentage?: number;
};

const PunchHole = styled.div<{ $bgColor: string }>`
  &:last-of-type {
    right: -30px;
    left: auto;
  }

  position: absolute;
  width: 40px;
  height: 22px;
  bottom: -10px;
  left: -30px;
  border-radius: 50%;
  z-index: 1;
  background-color: ${({ $bgColor }) => $bgColor};

  @media (min-width: ${breakpoints.m}) {
    width: 22px;
    height: 40px;
    top: -30px;
    right: -10px;
    bottom: auto;
    left: auto;

    &:last-of-type {
      top: auto;
      right: -10px;
      bottom: -30px;
    }
  }
`;

const Wrapper = styled.div<{ $state: TicketState }>`
  &:after {
    content: "";
    width: 100%;
    height: 40px;
    position: absolute;
    bottom: -40px;
    left: 0;
    background-color: inherit;
    border-radius: 0 0 10px 10px;
    border-top: 2px dashed var(--color-white);

    @media (min-width: ${breakpoints.m}) {
      width: 60px;
      height: 100%;
      top: 0;
      left: auto;
      bottom: auto;
      right: -60px;
      border-top: none;
      border-left: 2px dashed var(--color-white);
      border-radius: 0 10px 10px 0;
    }
  }

  ${({ $state }) => {
    switch ($state) {
      case "complete":
        return "background-color: #e1f5f3";
      case "error":
        return "background-color: #ffe1e1";
      case "incomplete":
      default:
        return "background-color: #e5f3fd";
    }
  }};
  box-sizing: border-box;
  border-radius: 10px 10px 0 0;
  margin: 0 0 60px 0;
  padding: var(--spacing-m) var(--spacing-m);
  position: relative;
  width: 100%;

  @media (min-width: ${breakpoints.m}) {
    margin: 0;
    width: calc(100% - 60px);
    height: calc(100% + 40px);
    border-radius: 10px 0 0 10px;
  }
`;

const Content = styled.div``;

const Title = styled(H1)`
  font-size: 1.75rem;
  font-family: var(--font-bold);
  font-weight: 700;
  margin: 0 0 var(--spacing-s) 0;
`;

const Subtitle = styled(H2)`
  margin: 0 0 var(--spacing-m) 0;
  font-size: 22px;
`;

const Duration = styled.div`
  display: flex;
  align-items: center;
  gap: var(--spacing-xs);
  margin-bottom: var(--spacing-xs);
  ${fontMedium}
`;

const Price = styled.div<{ $isFree: boolean }>`
  margin-top: var(--spacing-m);
  margin-bottom: var(--spacing-s);

  @media (min-width: ${breakpoints.m}) {
    margin-top: ${({ $isFree }) =>
      $isFree ? "var(--spacing-layout-xl)" : "var(--spacing-3-xl)"};
  }
`;

const PriceRow = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: var(--spacing-m);
  margin-bottom: var(--spacing-2-xs);
`;

const Ticket = ({
  state,
  title,
  subtitle,
  begin,
  end,
  isFree,
  bgColor = "var(--color-white)",
  reservationPrice,
  lowestPrice,
  highestPrice,
  priceUnit,
  taxPercentage,
}: Props): JSX.Element => {
  const { t } = useTranslation();

  const beginDate = t("common:dateWithWeekday", {
    date: begin && parseISO(begin),
  });

  const beginTime = t("common:timeWithPrefix", {
    date: begin && parseISO(begin),
  });

  const endDate = t("common:dateWithWeekday", {
    date: end && parseISO(end),
  });

  const endTime = t("common:time", {
    date: end && parseISO(end),
  });

  const reservationUnitPrices = {
    lowestPrice,
    highestPrice,
    priceUnit,
  };

  const duration = differenceInMinutes(new Date(end), new Date(begin));
  const timeString = `${beginDate} ${beginTime} - ${
    endDate !== beginDate ? endDate : ""
  }${endTime}`;

  const multiplier = getReservationVolume(duration, priceUnit);

  return (
    <Wrapper $state={state} data-testid="reservation__ticket--container">
      <PunchHole $bgColor={bgColor} />
      <Content>
        <Title data-test="reservation__title">{title}</Title>
        {subtitle && <Subtitle>{subtitle}</Subtitle>}
        {isFinite(duration) && (
          <Duration data-test="reservation__time-range">
            <IconCalendar />
            <span>
              {capitalize(timeString)}{" "}
              <NoWrap>
                (
                {t("common:duration", {
                  duration: formatDurationMinutes(duration),
                })}
                )
              </NoWrap>
            </span>
          </Duration>
        )}
        <Price $isFree={isFree} data-testid="reservation__price--container">
          {isFree ? (
            <Strongish>{t("reservationUnit:priceFree")}</Strongish>
          ) : (
            <>
              <PriceRow>
                <div>
                  <Strongish>
                    {t("reservations:reservationDuration", {
                      duration: formatDurationMinutes(duration),
                    })}
                  </Strongish>
                  {taxPercentage && (
                    <span>
                      (
                      {t("prices:taxPercentage", {
                        count: taxPercentage,
                      })}
                      )
                    </span>
                  )}
                </div>
                <div>
                  {reservationUnitPrices.highestPrice && (
                    <Strongish>
                      {priceUnit !== "FIXED" && `${multiplier} * `}
                      {getPrice(
                        reservationUnitPrices as ReservationUnitByPkType,
                        getPriceUnitMinutes(priceUnit),
                        true
                      )}
                    </Strongish>
                  )}
                </div>
              </PriceRow>
              <PriceRow>
                <div>
                  <Strongish>{t("prices:total")}</Strongish>
                </div>
                <div>
                  <Strongish>
                    {getReservationPrice(
                      reservationPrice,
                      i18n.t("prices:priceFree"),
                      i18n.language,
                      true
                    )}
                  </Strongish>
                </div>
              </PriceRow>
            </>
          )}
        </Price>
      </Content>
      <PunchHole $bgColor={bgColor} />
    </Wrapper>
  );
};

export default Ticket;
