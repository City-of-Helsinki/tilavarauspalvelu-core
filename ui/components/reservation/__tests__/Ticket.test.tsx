import * as React from "react";
import { render } from "../../../test/testUtils";
import Ticket from "../Ticket";

jest.mock("next/config", () => () => ({
  serverRuntimeConfig: {},
  publicRuntimeConfig: {},
}));

jest.mock("react-i18next", () => ({
  useTranslation: () => {
    return {
      t: (str: string) => str,
    };
  },
}));

jest.mock("next-i18next", () => ({
  i18n: { language: "fi", t: (str: string) => str },
}));

test("renders correctly", () => {
  const { asFragment } = render(
    <Ticket state="incomplete" title="Ticket title" />
  );
  expect(asFragment()).toMatchSnapshot();
});

test("renders correctly", () => {
  const { asFragment } = render(
    <Ticket state="incomplete" title="Ticket title" isFree />
  );
  expect(asFragment()).toMatchSnapshot();
});

test("renders correctly", () => {
  const { asFragment } = render(
    <Ticket
      state="complete"
      title="Backgstage huone"
      subtitle="Arabianrannan nuorisotalo"
      begin="2021-11-08T11:41:56.070Z"
      end="2021-11-08T13:41:56.070Z"
      isFree
      bgColor="var(--tilavaraus-gray)"
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
