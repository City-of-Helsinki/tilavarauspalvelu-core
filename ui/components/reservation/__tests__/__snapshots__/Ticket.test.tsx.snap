// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="Ticket__Wrapper-sc-16qon89-1 bxJppa"
    data-testid="reservation__ticket--container"
  >
    <div
      class="Ticket__PunchHole-sc-16qon89-0 kCZKkf"
    />
    <div
      class="Ticket__Content-sc-16qon89-2"
    >
      <h1
        class="typography__H1-sc-yczr8v-4 Ticket__Title-sc-16qon89-3 exespM cWvaak"
        data-test="reservation__title"
      >
        Ticket title
      </h1>
      <div
        class="Ticket__Price-sc-16qon89-6 iGKRoY"
        data-testid="reservation__price--container"
      >
        <div
          class="Ticket__PriceRow-sc-16qon89-7 jRyEGg"
        >
          <div>
            <span
              class="typography__Strongish-sc-yczr8v-1 pEndj"
            >
              reservations:reservationDuration
            </span>
          </div>
          <div />
        </div>
        <div
          class="Ticket__PriceRow-sc-16qon89-7 jRyEGg"
        >
          <div>
            <span
              class="typography__Strongish-sc-yczr8v-1 pEndj"
            >
              prices:total
            </span>
          </div>
          <div>
            <span
              class="typography__Strongish-sc-yczr8v-1 pEndj"
            >
              prices:priceFree
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Ticket__PunchHole-sc-16qon89-0 kCZKkf"
    />
  </div>
</DocumentFragment>
`;

exports[`renders correctly 2`] = `
<DocumentFragment>
  <div
    class="Ticket__Wrapper-sc-16qon89-1 ktMrPx"
    data-testid="reservation__ticket--container"
  >
    <div
      class="Ticket__PunchHole-sc-16qon89-0 kCZKkf"
    />
    <div
      class="Ticket__Content-sc-16qon89-2"
    >
      <h1
        class="typography__H1-sc-yczr8v-4 Ticket__Title-sc-16qon89-3 exespM cWvaak"
        data-test="reservation__title"
      >
        Ticket title
      </h1>
      <div
        class="Ticket__Price-sc-16qon89-6 iGKRoY"
        data-testid="reservation__price--container"
      >
        <div
          class="Ticket__PriceRow-sc-16qon89-7 jRyEGg"
        >
          <div>
            <span
              class="typography__Strongish-sc-yczr8v-1 pEndj"
            >
              reservations:reservationDuration
            </span>
          </div>
          <div />
        </div>
        <div
          class="Ticket__PriceRow-sc-16qon89-7 jRyEGg"
        >
          <div>
            <span
              class="typography__Strongish-sc-yczr8v-1 pEndj"
            >
              prices:total
            </span>
          </div>
          <div>
            <span
              class="typography__Strongish-sc-yczr8v-1 pEndj"
            >
              prices:priceFree
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Ticket__PunchHole-sc-16qon89-0 kCZKkf"
    />
  </div>
</DocumentFragment>
`;

exports[`renders correctly 3`] = `
<DocumentFragment>
  <div
    class="Ticket__Wrapper-sc-16qon89-1 ktMrPx"
    data-testid="reservation__ticket--container"
  >
    <div
      class="Ticket__PunchHole-sc-16qon89-0 kCZKkf"
    />
    <div
      class="Ticket__Content-sc-16qon89-2"
    >
      <h1
        class="typography__H1-sc-yczr8v-4 Ticket__Title-sc-16qon89-3 exespM cWvaak"
        data-test="reservation__title"
      >
        Ticket title
      </h1>
      <div
        class="Ticket__Price-sc-16qon89-6 hBNSGZ"
        data-testid="reservation__price--container"
      >
        <span
          class="typography__Strongish-sc-yczr8v-1 pEndj"
        >
          reservationUnit:priceFree
        </span>
      </div>
    </div>
    <div
      class="Ticket__PunchHole-sc-16qon89-0 kCZKkf"
    />
  </div>
</DocumentFragment>
`;

exports[`renders correctly 4`] = `
<DocumentFragment>
  <div
    class="Ticket__Wrapper-sc-16qon89-1 cQAWtY"
    data-testid="reservation__ticket--container"
  >
    <div
      class="Ticket__PunchHole-sc-16qon89-0 hgEUEW"
    />
    <div
      class="Ticket__Content-sc-16qon89-2"
    >
      <h1
        class="typography__H1-sc-yczr8v-4 Ticket__Title-sc-16qon89-3 exespM cWvaak"
        data-test="reservation__title"
      >
        Backgstage huone
      </h1>
      <h2
        class="typography__H2-sc-yczr8v-5 Ticket__Subtitle-sc-16qon89-4 dQVsnC cdeDAn"
      >
        Arabianrannan nuorisotalo
      </h2>
      <div
        class="Ticket__Duration-sc-16qon89-5 eGJEwH"
        data-test="reservation__time-range"
      >
        <svg
          class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
          role="img"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <path
              d="M0 0h24v24H0z"
            />
            <path
              d="M17 2a1 1 0 011 1v1h4v17H2V4h4V3a1 1 0 112 0v1h8V3a1 1 0 011-1zm3 9H4v8h16v-8zm0-5H4v3h16V6z"
              fill="currentColor"
            />
          </g>
        </svg>
         Common:dateWithWeekday common:timeWithPrefix - common:time (common:duration)
      </div>
      <div
        class="Ticket__Price-sc-16qon89-6 hBNSGZ"
        data-testid="reservation__price--container"
      >
        <span
          class="typography__Strongish-sc-yczr8v-1 pEndj"
        >
          reservationUnit:priceFree
        </span>
      </div>
    </div>
    <div
      class="Ticket__PunchHole-sc-16qon89-0 hgEUEW"
    />
  </div>
</DocumentFragment>
`;
